modxcms/revolution
# Fix Static File Creation Issue

### What does it do?

Corrected the file path reference used to create new static element files.

### Why is it needed?

Two connected problems are resolved here:
1. The logging issue referenced in issue #16005
2. Without this fix, a duplicate file (which is empty) is produced within the selected media source (but having the full server path). For example, if choosing the "Filesystem" media source and creating a static chunk at "assets/chunks/static-chunk-1.html", two files will be created (assuming an example server root path of "/var/www/site"):
-  assets/chunks/static-chunk-1.html (as expected)
-  assets/chunks/var/www/site/assets/chunks/static-chunk-1.html (empty duplicate at wrong location)

### How to Test

1. To verify the second problem above, create a static element or two on any other install/branch of 3.x. Take a look at your logs and the directories/files that get created.
2. Load _this_ PR branch and create one of each element with a static file source specified. For each, be sure to enter a little text/code into the element's code field to verify its content is actually saved to the target file. 
3. Monitor your manager error log to verify that entries do not get created when a static element is successfully created and saved. 
4. Verify in your file structure that there is no duplication of the static file in another area of the file system (see item 2 above).
5. Although probably not necessary (because I don't believe this code intersects with the parsing end of things), create a page that makes use of each static element to ensure the output is as expected.

### Additional Note

This PR does not seek to fix either of these two issues:
1. If you specify a file type that is not listed in the `upload_files` system setting  you will be unable to save the static element. A resolution for this, perhaps by implementing a new system setting for allowable static types and/or the ability to specify allowable types on a media source basis, will be proposed in a separate PR.
2. Selecting "None" as the media source currently results in unexpected behavior and absolute paths do not work (see also #15927). This will be addressed in a separate PR as well.

### Related issue(s)/PR(s)
Resolves #16005
